ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/transaction/BesuPrivateNonceProvider.java

```diff
@@ -12,54 +12,42 @@
  */
 package tech.pegasys.ethsigner.core.requesthandler.sendtransaction.transaction;
 
+import tech.pegasys.ethsigner.core.jsonrpc.JsonRpcRequest;
+import tech.pegasys.ethsigner.core.jsonrpc.JsonRpcRequestId;
 import tech.pegasys.ethsigner.core.requesthandler.sendtransaction.NonceProvider;
 
-import java.io.IOException;
 import java.math.BigInteger;
 
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
-import org.web3j.protocol.besu.Besu;
-import org.web3j.protocol.core.Request;
-import org.web3j.protocol.core.methods.response.EthGetTransactionCount;
 import org.web3j.utils.Base64String;
 
 public class BesuPrivateNonceProvider implements NonceProvider {
 
-  private static final Logger LOG = LogManager.getLogger();
-
-  private final Besu besu;
+  protected static int lastId = 0;
+  private final VertxNonceRequestTransmitter vertxNonceRequestTransmitter;
   private final String accountAddress;
   private final Base64String privacyGroupId;
 
-  BesuPrivateNonceProvider(
-      final Besu besu, final String accountAddress, final Base64String privacyGroupId) {
-    this.besu = besu;
+  public BesuPrivateNonceProvider(
+      final String accountAddress,
+      final Base64String privacyGroupId,
+      final VertxNonceRequestTransmitter vertxNonceRequestTransmitter) {
+    this.vertxNonceRequestTransmitter = vertxNonceRequestTransmitter;
     this.accountAddress = accountAddress;
     this.privacyGroupId = privacyGroupId;
   }
 
   @Override
   public BigInteger getNonce() {
-    return getNonceFromClient();
+    final JsonRpcRequest request = generateRequest();
+    return vertxNonceRequestTransmitter.requestNonce(request);
   }
 
-  private BigInteger getNonceFromClient() {
+  protected JsonRpcRequest generateRequest() {
+    final JsonRpcRequest request = new JsonRpcRequest("2.0", "priv_getEeaTransactionCount");
+    request.setParams(new Object[] {accountAddress, privacyGroupId});
+    lastId += 1;
```

Comment:
nit: lastId++ ??

Reply:
resolved in VertxNonceRequestTransmitter (by using AtomicInteger)